app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -156,11 +157,13 @@ export default class ComposerService extends Service {
     "_disableSubmit"
   )
   get disableSubmit() {
-    return (
+    return applyValueTransformer(
+      "composer-disable-submit",
       this.model?.loading ||
-      this.isUploading ||
-      this.isProcessingUpload ||
-      this._disableSubmit
+        this.isUploading ||
+        this.isProcessingUpload ||
+        this._disableSubmit,
+      { model: this.model }
```

Comment:
🏌️ – maybe we could, instead of only passing the `model`,  pass along the `this` as well :
 `{composerService: this, composerModel: this.model}`


What do you think? I wonder if, in this case, we just want to pass the model; we could name it `composer,` the usage would be :cheefs_kiss:

```js
  api.registerValueTransformer(
    "composer-disable-submit",
   ({ value, context: { composer } }) => {
      
    }
  );
```

Reply:
IMO passing along `this` is a code smell. Now that you mention it, given we are within the context of the composer service I wonder if it would be best to not pass anything as a context. I like the simplicity of expecting that the developer will look up the composer service from the api initializer.